statsmodels/stats/robustros.py

```diff
@@ -0,0 +1,477 @@
+            self.data['modeled_data'] = self.transform_out(
+                slope*self.data['Zprelim'][nondet_selector] + intercept
+            )
+
+            # select out the final data
+            self.data['modeled'] = self.data.apply(_select_modeled, axis=1)
+
+        # create the debug attribute as a copy of the self.data attribute
+        self.debug = self.data.copy(deep=True)
+
+        # select out only the necessary columns for data
+        final_cols = ['modeled', self.result, self.censorship]
+        self.data = self.data[final_cols]
+
+    def plot(self, ax=None, show_raw=True, raw_kwds={}, model_kwds={},
+             leg_kwds={}, ylog=True):
+        '''
+        Generate a QQ plot of the raw (censored) and modeled data.
+
+        Parameters
+        ----------
+        ax : optional matplotlib Axes
+            The axis on which the figure will be drawn. If no specified
+            a new one is created.
+        show_raw : optional boolean (default = True)
+            Toggles on (True) or off (False) the drawing of the censored
+            quantiles.
+        raw_kwds : optional dict
+            Plotting parameters for the censored data. Passed directly
+            to `ax.plot`.
+        model_kwds : optional dict
+            Plotting parameters for the modeled data. Passed directly to
+            `ax.plot`.
+        leg_kwds : optional dict
+            Optional kwargs for the legend, which is only drawn if
+            `show_raw` is True. Passed directly to `ax.legend`.
+        ylog : optional boolean (default = True)
+            Toggles the logarthmic scale of the y-axis.
+
+        Returns
+        -------
+        ax : matplotlib Axes
+
+        '''
+
+        fig, ax = utils.create_mpl_ax(ax)
+
+        # legend options
+        leg_params = {
+            'loc': 'upper left',
+            'fontsize': 8
+        }
+        leg_params.update(leg_kwds)
+
+        # modeled data
+        mod_symbols = {
+            'marker': 'o',
+            'markersize': 6,
+            'markeredgewidth': 1.0,
+            'markeredgecolor': 'blue',
+            'markerfacecolor': 'blue',
+            'linestyle': 'none',
+            'label': 'Modeled data',
+            'alpha': 0.87
+        }
+        mod_symbols.update(model_kwds)
+        osm_mod, osr_mod = stats.probplot(
+            self.data['modeled'], fit=False
+        )
+        ax.plot(osm_mod, osr_mod, **mod_symbols)
+
+        # raw data
+        if show_raw:
+            raw_symbols = {
+                'marker': 's',
+                'markersize': 6,
+                'markeredgewidth': 1.0,
+                'markeredgecolor': '0.35',
+                'markerfacecolor': 'none',
+                'linestyle': 'none',
+                'label': 'Censored data',
+                'alpha': 0.70
+            }
+            raw_symbols.update(raw_kwds)
+            osm_raw, osr_raw = stats.probplot(
+                self.data[self.result], fit=False
+            )
```

Comment:
are the locations of closing braces and brackets because of automatic formatting or is there a style convention somewhere for this?